### PR TITLE
Test cmd cancel

### LINKED ISSUE #343: work cancel <unit_id> doesn't cancel if running on another node, gives exit code of 0
I have a receptor cluster of 3 machines, call them A, B, and C

I have a job running on node A, and I know the unit id.

I shell into node B and run `receptorctl work cancel <unit_id>`. Then I get the standard out of:

```
Cancelled: ('wFvhMc80',)
```

it also gives an exit code of 0.

From reading this, I would expect that the unit id `wFvhMc80` has been canceled, but it has not been.

If, instead, I shell into node A and run the same command, it does cancel it.

### PR DESCRIPTION
**Pull request by thenets**
Fixes https://github.com/ansible/receptor/issues/343:
- https://github.com/ansible/receptor/issues/343

### CODE REVIEW

File: receptorctl/tests/test_cli.py [hunk 1]
```diff
@@ -25,3 +49,38 @@ def test_cmd_status(self, invoke_as_json):
...
+        sleep_for = 5 # in seconds
+
+        work = node1_controller.submit_work("sleep", str(sleep_for), node="node3")
+        result = work.pop("result")
+        unitid = work.pop("unitid")
+
+        assert result == "Job Started"
+        assert wait_for_work_finished(node1_controller, unitid, (sleep_for + 5)), "Workunit timed out and never finished"
+
+        node1_controller.get_work_results(unitid)
```
Comment (thenets) [inline receptorctl/tests/test_cli.py]: I'll need help with this. I don't know if it's a bug or a feature (no joke), but I'm getting an error here:

```python
assert wait_for_work_finished(node1_controller, unitid, (sleep_for + 5)), "Workunit timed out and never finished"
    
>       node1_controller.get_work_results(unitid)

tests[/test_cli.py:71](): 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <receptorctl.socket_interface.ReceptorControl object at 0x7f40e099c520>
unit_id = '2jUdu2xn', return_socket = False, return_sockfile = True

    def get_work_results(self, unit_id, return_socket=False, return_sockfile=True):
        self.connect()
        self.writestr(f"work results {unit_id}\n")
        text = self.readstr()
        m = re.compile("Streaming results for work unit (.+)").fullmatch(text)
        if not m:
            errmsg = "Failed to get results"
            if str.startswith(text, "ERROR: "):
                errmsg = errmsg + ": " + text[7:]
>           raise RuntimeError(errmsg)
E           RuntimeError: Failed to get results: could not verify signature: signature is empty

receptorctl[/socket_interface.py:255](): RuntimeError
```

**Comment by thenets** [comment]
Was found an error with the TCP connection.
I'll migrate the code to use file sockets only.

File: receptorctl/tests/test_workunit.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        state_unitid = work.pop("unitid")
+        assert state_result == "Job Started"
+
+        # Run and check cancel command
+        cancel_output = node1_controller.simple_command(f"work cancel {state_unitid}")
+        assert cancel_output["cancelled"] == state_unitid
+
+        # Get work list and check for the workunit detail state
+        work_list = node1_controller.simple_command(f"work list")
+        assert work_list[state_unitid]["Detail"] == "Killed"
```
Comment (thenets) [inline receptorctl/tests/test_workunit.py:164]: @fosterseth I think that I'm doing something wrong. I'm getting the following error:

```java
./tests/test_workunit.py::TestWorkUnit::test_workunit_cmd_cancel
Failed: [undefined]AssertionError: assert 'Starting Worker' == 'Killed'
  - Killed
  + Starting Worker
```

And even if I sleep and wait for a state change, I get the following error:

```java
./tests/test_workunit.py::TestWorkUnit::test_workunit_cmd_cancel
Failed: [undefined]AssertionError: assert 'Starting Worker' == 'Launching command runner'
  - Killed
  + Launching command runner
```

File: receptorctl/tests/test_cli.py (hunk 1 above)
Comment (thenets) [reply receptorctl/tests/test_cli.py]: Was found an error with the TCP connection.
I'll migrate the code to use file sockets only.

File: receptorctl/tests/test_workunit.py (hunk 2 above)
Comment (thenets) [reply receptorctl/tests/test_workunit.py:164]: Adding the blocked tag until we investigate this error.

Comment (thenets) [reply receptorctl/tests/test_workunit.py:164]: If I cancel the `workunit` less than 1 second after creating it, sometimes the workunit stays forever in the state `Launching command runner`.

**Comment by thenets** [comment]
I believe it's ready for review. When approved, I'll squash and merge it.
EDIT: I forgot about the auto-merge. I'll squash now.
EDIT 2: squashed.

**Comment by thenets** [comment]
Blocked. The pipeline `CI / container (pull_request)` is broken and it's not related to my PR.
EDIT: I was wrong. Was my fault.

File: receptorctl/test-requirements.txt [hunk 3]
```diff
@@ -1,6 +1,8 @@
 -r build-requirements.txt
-pytest==6.1.1
+-e .
```
Comment (shanemcd) [inline receptorctl/test-requirements.txt]: I'm not sure why you needed to add this line but the tests work fine for me when deleting it and it also fixes the container build.

Comment (thenets) [reply receptorctl/test-requirements.txt]: I think you already told me to not do that. I didn't realize that this file was being read.
Thank you for investigating that!

**Review by shanemcd — APPROVED** [review]

**Pull request merged**